organization/search_indexes.py

```diff
@@ -0,0 +1,205 @@
+        'organization_end_date_dt': first_cited,
+        'organization_adminlevel1_ss': list(admin_l1_names),
+        'text': content
+    }
+    '''
+
+    CONTENT_FIELDS = (
+        'name',
+        'aliases',
+        'classifications',
+        'headquarters',
+        'parent_names',
+        'countries',
+        'exact_locations',
+        'areas',
+    )
+
+    open_ended = indexes.CharField()
+    name = indexes.CharField()
+    parent_names = indexes.MultiValueField(faceted=True)
+    memberships = indexes.MultiValueField(faceted=True)
+    classifications = indexes.MultiValueField(faceted=True)
+    aliases = indexes.MultiValueField()
+    headquarters = indexes.CharField()
+    exact_locations = indexes.MultiValueField()
+    areas = indexes.MultiValueField()
+    adminlevel1s = indexes.MultiValueField(faceted=True)
+
+    def get_model(self):
+        # For some reason, Haystack doesn't want to discover the index if we
+        # import the model into the global namespace for this module. So,
+        # import it here.
+        from organization.models import Organization
+
+        return Organization
+
+    def prepare(self, object):
+        self.prepared_data = super().prepare(object)
+
+        self.prepared_data['content'] = self._prepare_content(self.prepared_data)
+        self.prepared_data['countries'] = self._prepare_countries(self.prepared_data)
+
+        return self.prepared_data
+
+    def _prepare_countries(self, prepared_data):
+        countries = set()
+
+        for division in prepared_data['division_ids']:
+            countries.update([country_name(division)])
+
+        return list(countries)
+
+    def prepare_published(self, object):
+        parents = object.parent_organization.all()
+
+        return all([p.value.published for p in parents])
+
+    def prepare_division_ids(self, object):
+        division_ids = set()
+
+        # Start by getting the division ID recorded for this org
+        org_division_id = object.division_id.get_value()
+        if org_division_id:
+            division_ids.update([org_division_id.value])
+
+        # Grab foreign key sets
+        emplacements = object.emplacementorganization_set.all()
+
+        for emp in emplacements:
+            emp = emp.object_ref
+            site = emp.site.get_value()
+
+            if site:
+                exactloc_name = site.value.name
+                emp_division_id = site.value.division_id
+
+                if emp_division_id:
+                    division_ids.update([emp_division_id])
+
+        return list(division_ids)
+
+    def prepare_start_date(self, object):
+        return self._format_date(object.firstciteddate.get_value())
+
+    def prepare_end_date(self, object):
+        return self._format_date(object.lastciteddate.get_value())
+
+    def prepare_open_ended(self, object):
+        open_ended = object.open_ended.get_value()
+
+        if open_ended:
+            return open_ended.value
```

Comment:
so we want to return `open.ended.value` or `None`? what is the false-y value of open_ended?

Reply:
`get_value()` [returns the field or `None`](https://github.com/security-force-monitor/complex_fields/blob/328aef79a458fb568f9917929f9817c8c73f729c/complex_fields/models.py#L142). So, `None`.